pandas/core/strings.py

```diff
@@ -1815,36 +1817,34 @@ def __init__(self, data):
     def _validate(data):
         from pandas.core.index import Index
 
-        if (isinstance(data, ABCSeries) and
-                not ((is_categorical_dtype(data.dtype) and
-                      is_object_dtype(data.values.categories)) or
-                     (is_object_dtype(data.dtype)))):
-            # it's neither a string series not a categorical series with
-            # strings inside the categories.
-            # this really should exclude all series with any non-string values
-            # (instead of test for object dtype), but that isn't practical for
-            # performance reasons until we have a str dtype (GH 9343)
+        if isinstance(data, ABCMultiIndex):
+                raise AttributeError('Can only use .str accessor with Index, '
+                                     'not MultiIndex')
+
+        # see src/inference.pyx which can contain string values
+        allowed_types = ['string', 'unicode', 'empty',
+                         'mixed', 'mixed-integer']
+        if isinstance(data, ABCSeries):
+            allowed_types = allowed_types + ['bytes']
+
+        values = data if isinstance(data, Index) else data.values
+        if is_categorical_dtype(data.dtype):
+            inf_type = lib.infer_dtype(values.categories)
```

Comment:
pls pls don't use abbreviation. ``inferred_type`` is the name for this

Reply:
This came from the existing code. Will change